core/ledger/snapshotgrpc/snapshot_service.go

```diff
@@ -84,6 +103,33 @@ func (s *SnapshotService) QueryPendings(ctx context.Context, signedRequest *pb.S
 	return &pb.QueryPendingSnapshotsResponse{BlockNumbers: result}, nil
 }
 
+func (s *SnapshotService) checkACL(channelID string, resName string, signatureHdr *cb.SignatureHeader, signedRequest *pb.SignedSnapshotRequest) error {
+	if signatureHdr == nil {
+		return errors.New("missing signature header")
+	}
+
+	expirationTime := crypto.ExpiresAt(signatureHdr.Creator)
+	if !expirationTime.IsZero() && time.Now().After(expirationTime) {
```

Comment:
@ale-linux Thanks for your comment. I confirm that the events for which the ACL is checked for snapshot service are never on orderer events or on commit path.

Reply:
I did not notice this expiration check on other APIs that allow admin operations (e.g., join channel). Better not to have a different behavior for snapshot related APIs.

Reply:
+1 @manish-sethi, good point

Reply:
@manish-sethi The other APIs that allow admin ops are done via endorsor ProcessProposal that checks expiration via an auth filter [expiration.go](https://github.com/hyperledger/fabric/blob/master/core/handlers/auth/filter/expiration.go#L49) Therefore, individual cmd doesn't need separate call to check certification expiration. However, because snapshot service is a separate grpc service, we should add this check.

Reply:
@manish-sethi The other APIs that allow admin ops are done via endorsor ProcessProposal that checks expiration via an auth filter expiration.go Therefore, individual cmd doesn't need separate call to check certificate expiration. However, because snapshot service is a separate grpc service, we should add this check.

Reply:
I see, I missed that part here. Thanks for the clarification @wenjianqiao

Reply:
Just to take a note, ideally, that should also have been performed at one place like ACLCheck but not related to this PR.